neynar-webhook-kafka-consumer/src/index.ts

```diff
@@ -0,0 +1,276 @@
+import {
+  Kafka,
+  Consumer,
+  EachMessagePayload,
+  KafkaMessage,
+  logLevel,
+} from "kafkajs";
+import { createClient } from "redis";
+import { v4 as uuidv4 } from "uuid";
+import dotenv from "dotenv";
+
+import { FarcasterEvent } from "./types";
+
+dotenv.config();
+
+if (!process.env.KAFKA_BROKERS) {
+  console.error("KAFKA_BROKERS environment variable is required");
+  process.exit(1);
+}
+
+if (!process.env.KAFKA_USERNAME) {
+  console.error("KAFKA_USERNAME environment variable is required");
+  process.exit(1);
+}
+
+if (!process.env.KAFKA_PASSWORD) {
+  console.error("KAFKA_PASSWORD environment variable is required");
+  process.exit(1);
+}
+
+if (!process.env.KAFKA_CONSUMER_GROUP) {
+  console.error("KAFKA_CONSUMER_GROUP environment variable is required");
+  process.exit(1);
+}
+
+const is_production = process.env.NODE_ENV === "production";
+
+const KAFKA_USERNAME = process.env.KAFKA_USERNAME;
+const KAFKA_PASSWORD = process.env.KAFKA_PASSWORD;
+const brokers = process.env.KAFKA_BROKERS
+  ? process.env.KAFKA_BROKERS.split(",")
+  : ["localhost:9092"];
+
+// https://kafka.js.org/docs/configuration
+const kafka = new Kafka({
```

Comment:
we should probably add a `retry` block to this, too. Your reconnect logic will reconnect at the start, but if it dies while the app is running I think it will exit and and have to start again.

Reply:
i just checked the docs and the retry code actually defaults to being enabled. lets trust their defaults